background/adapter/deluge.js

```diff
@@ -0,0 +1,57 @@
+if (typeof torrentToWeb === 'undefined') {
+    var torrentToWeb = {};
+};
+
+if (typeof torrentToWeb.adapter === 'undefined') {
+    torrentToWeb.adapter = {};
+};
+
+torrentToWeb.adapter.deluge = function(baseUrl, username, password, autostart)
+{
+    var baseUrlObject = new URL(baseUrl);
+    baseUrlObject.username = username;
+    baseUrlObject.password = password;
+    baseUrlObject.pathname = '/json';
+    baseUrlObject.search = '';
+    baseUrlObject.hash = '';
+
+    baseUrl = baseUrlObject.toString();
+
+    return {
+        send: function(filename, data, callback)
+        {
+            var fileReader = new FileReader();
+            fileReader.addEventListener('load', function(){
+                var requestData = {
+                    'id': '2',
+                    'method': 'core.add_torrent_file',
+                    'params': [filename,window.btoa(fileReader.result), null]
+                };
+
+                sendRequest(requestData, callback, null);
+            });
+            fileReader.readAsBinaryString(data);
+        }
+    };
+
+    function sendRequest(requestData, callback, delugeSessionId){
```

Comment:
Is the session ID actually used anywhere? It doesn't seam like you re-call `sendRequest()` in case of failure. If there is no actual session request going on, consider getting rid of this method (like in the rutorrent adapter).